magento/magento2
# magento/magento2#35810: Integrity constraint violation error when saving product without sku

Fix saving product without sku.





### Description (*)

Change sku value of object from null to '' in method \Magento\Catalog\Model\Product\Attribute\Backend\Sku::beforeSave to prevent mistake SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'sku' cannot be null.

### Related Pull Requests

--

### Fixed Issues (if relevant)

1. Fixes magento/magento2#35810

### Manual testing scenarios (*)

1. Admin - Stores - Attributes - Product - Search for sku - edit - set "Values Required" - No and save
2. Clear cache
3. Create product without sku - save.

Before fix:
Invalid error message displayed to the user.
Error in 2.4-develop branch: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'sku' cannot be null, query was: INSERT INTO catalog_product_entity (entity_id, attribute_set_id, type_id, sku, has_options, required_options, created_at, updated_at) VALUES (?, ?, ?, ?, ?, ?, ?, ?)

After fix:
Invalid error/exception should not be displayed.
Product successfully created.

### Questions or comments


### Contribution checklist (*)
